### ISSUE TITLE
`function-calc-no-unspaced-operator` rule whether to consider adding regular filtering options

### ISSUE THREAD

**Issue by jiangmaniu** [stylelint/stylelint#5517]
> Clearly describe the bug

Vue officially released the [v-bind grammar](https://v3.vuejs.org/api/sfc-style.html#state-driven-dynamic-css) in [v3.2.2](https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md#320-2021-08-09) to support js and facilitate the use of js variables in css. This grammar will conflict with the `function-calc-no-unspaced-operator` rule in calc(). I checked this The configured rule has only a switch. If you want the editor not to report an error, you can only close the rule globally or use the Ignoring code. I personally feel that v-bind will be used frequently in future vue projects. The above two methods are not very good solutions. Maybe you can add a configuration item to filter some regular or string in the rule.

> Which rule, if any, is the bug related to?

`function-calc-no-unspaced-operator`

> What code is needed to reproduce the bug?

```css
.box {
      height: calc(100% - v-bind('cssVars.layoutHeaderHeight'));
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-calc-no-unspaced-operator":  true
  }
}
```

> Which version of stylelint are you using?

v13.13.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

e.g. "CLI with `stylelint "**/*.css" --config myconfig.json`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Related to v-bind syntax of vue v3.2.x

> What did you expect to happen?

e.g. "No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

![image](https://user-images.githubusercontent.com/50100681/132305715-507b677a-858e-4262-818c-b126facb4efd.png)

![image](https://user-images.githubusercontent.com/50100681/132306731-5f721caf-1eee-46f4-b8e8-cb6c4851ea21.png)

**Comment by jeddy3** [comment]
@jiangmaniu Thanks for the report and for using the template.

I can reproduce [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilAjANhGANYC0AzgJYBeM8KYS+YAFAIwAMHApCqX6QC2AT1JgAruQAuEQaQBm4zGCmUImdgCcYggJS6A3FgC+WEABoQ5YZilIAHgnDlyF8OvmUA5k4zYU6CCa4vgw5IH0fjg4gYrKqupiTGCkmBCkSuQADkhgMFCkEFkwmkgymhEoUsEwWDimmMYgxkA) with vanilla CSS:

```css
a {
  block-size: calc(100% - --my-custom-function(1rem));
}
```

Refactoring the rule to use [the value parser](https://www.npmjs.com/package/postcss-value-parser), rather than `style-search`, will likely fix the issue (or at least make it easier to fix). [Here's an example a similar refactor](https://github.com/stylelint/stylelint/pull/5256/files).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

Please branch of `v14` rather than `master`.

**Comment by lachieh** [comment]
I'd love to pick this one up, but I'm hoping for a few more pointers. I see that contents of the `style-search` callback makes up a good chunk of the code. Would it be best to take that code out and re-implement with `valueParser` or try to change as little as possible?

**Comment by lachieh** [comment]
I saw that commit mentioned in the last issue and the example you linked looks like it has replaced 80% of the original code. I'm just looking for a bit more guidance. Sorry to be an inconvenience. Thanks in advance!

**Comment by lachieh** [comment]
This is fantastic, thank you for this! I'll comment with any questions.
